x/bank/types/key.go

```diff
@@ -37,8 +38,16 @@ func DenomMetadataKey(denom string) []byte {
 // AddressFromBalancesStore returns an account address from a balances prefix
 // store. The key must not contain the perfix BalancesPrefix as the prefix store
 // iterator discards the actual prefix.
+//
+// If invalid key is passed, AddressFromBalancesStore returns InvalidKey.
 func AddressFromBalancesStore(key []byte) sdk.AccAddress {
+	if len(key) == 0 {
+		return sdk.AccAddress(InvalidKey)
```

Comment:
This seems like downstream consumers need to check if `addr == sdk.AccAddress(InvalidKey)` everywhere. Isn't returning an error here more correct?

I'm okay to break the function signature here

Reply:
@amaurym, I think API breakages are out of scope might be out of scope for the security changes, otherwise we now also have go change all call sites, make API breakage announcements. If it is trivial @cuonglm, but otherwise, I'd kindly ask that perhaps an issue be filed for that as auditing and fuzzing code is an intense task of its own.

Reply:
I would keep this fix for backporting, then we can break API in next release.

Reply:
Here's the code on 0.42.x, there's nothing to backport:
https://github.com/cosmos/cosmos-sdk/blob/7648bfca45b9d0897103ec739210607dce77c4fb/x/bank/types/key.go#L39-L46

We're about to release 0.43 in the next couple of weeks, so imo right now is the best time to introduce breaking API changes. If you don't want introduce breaking changes, maybe panicking (like 0.42) is better?

I just don't think introducing a new concept of `sdk.AccAddress(0x00)` is desirable.

Reply:
Ah right, so breaking the API seems feasible.